entc/gen/type.go

```diff
@@ -680,21 +680,29 @@ func (t Type) RelatedTypes() []*Type {
 	return related
 }
 
-// check checks the schema type.
-func (t *Type) check() error {
-	pkg := t.Package()
-	if token.Lookup(pkg).IsKeyword() {
-		return fmt.Errorf("schema lowercase name conflicts with Go keyword %q", pkg)
+// CheckNameConflicts will determine if a name is going to conflict with any
+// pre-defined names
+func CheckNameConflicts(name string) error {
```

Comment:
Maybe use something like `ValidSchemaName` / `CheckSchemaName` / `CheckIdent`? 
```suggestion
func ValidSchemaName(name string) error {
```